### PR TITLE
[TVMScript] Use op attribute to control whether to print dtype in TVMScript

### PR DESCRIPTION
**Pull request by liangW-intellif**
This PR adds an op attribute `TScriptDtypePrintLocation`, and modifies the dtype printing logic of the builtin op to check this attribute. So that user defined operators can use it to specify how there dtype argument are printed by appending attributes instead of appending members to `dtype_first_arg`/`dtype_last_arg`.
@junrushao @wrongtest-intellif

### CODE REVIEW

**Comment by liangW-intellif** [comment]
@tvm-bot rerun

File: src/script/printer/tir/expr.cc [hunk 1]
```diff
@@ -222,23 +222,9 @@ TVM_STATIC_IR_FUNCTOR(IRDocsifier, vtable)
...
-          tir::builtin::mma_fill().get(),
-          tir::builtin::vectorlow().get(),
-          tir::builtin::vectorhigh().get(),
-          tir::builtin::vectorcombine().get(),
-          Op::Get("tir.type_annotation").get(),
-      };
+      static const OpAttrMap<tir::TScriptDtypePrintLocation> dtype_locations =
+          Op::GetAttrMap<tir::TScriptDtypePrintLocation>("TScriptDtypePrintLocation");
+      tir::ScriptDtypePrintLocation dtype_print_location = tir::ScriptDtypePrintLocation::kNone;
       static const std::unordered_set<const Object*> dtype_last_arg = {
```
Comment (MasterJH5574) [inline src/script/printer/tir/expr.cc:228]: Thanks for the improvement! I suppose the `dtype_last_arg` can be smoothly removed right?

Comment (junrushao) [reply src/script/printer/tir/expr.cc:228]: @liangW-intellif we should remove `dtype_last_arg` given it's not used any more

Comment (liangW-intellif) [reply src/script/printer/tir/expr.cc]: Done, thanks for reminding.

**Review by MasterJH5574 — APPROVED** [review]

**Review by Hzfengsy — APPROVED** [review]

**Review by junrushao — APPROVED** [review]

**Pull request merged**